tests/e2e/cucumber/step_definitions/app.files.ts

```diff
@@ -394,3 +392,45 @@ When(
     }
   }
 )
+
+When(
+  'the public accesses the last created public link with password {string}',
+  async function (this: World, password: string): Promise<void> {
+    const publicLinkPage = await PublicLinkPage.setup()
+    await publicLinkPage.navigateToPublicLink()
+    await publicLinkPage.authenticatePassword(password)
+  }
+)
+
+Then(
+  /the public should( not | )see the following (resource|resources)$/,
+  async function (this: World, actionType: string, _: string, stepTable: DataTable): Promise<void> {
+    await filesCta.checkResourcesExistence({ actionType, stepTable })
+  }
+)
+
+When(
+  'the public uploads the following files on the files-drop page',
+  async function (this: World, stepTable: DataTable): Promise<void> {
+    const publicLinkPage = new PublicLinkPage()
+
+    const files = stepTable.raw().map((f) => f[0])
+    for (const file of files) {
+      const uploadInfo = this.filesEnvironment.getFile({ name: file })
+      await publicLinkPage.uploadFiles(uploadInfo.path)
+    }
+  }
+)
+
+Then(
+  /the public should( not | )see the following files on the files-drop page$/,
+  async function (this: World, actionType: string, stepTable: DataTable): Promise<void> {
+    const resourceType = 'uploaded'
+    await filesCta.checkResourcesExistence({ actionType, stepTable, resourceType })
```

Comment:
resourceType is not suitable for indicating type of page
```suggestion
    await filesCta.checkResourcesExistence({ actionType, stepTable, pageType: 'upload' })
```